### DISCUSSION TITLE
Increase memory limit for Edge functions as part of upgrading project compute size

### DISCUSSION THREAD

**Discussion by mediashane** [supabase/supabase#20558] [Feature Requests]
**The requested feature is: allow upgrading project compute size to also increase the memory limit on Edge functions**

There is a currently a limitation on the amount of memory an Edge function can use:

_Memory limit for the function: 150 MB (heap) + 150 MB (external - array buffers/ WASM)_

This is a smallish amount of memory, and doesn't allow for a successful API call to OpenAI's Whisper model:

    const transcription = await openai.audio.transcriptions.create({
      file: audioData,
      model: 'whisper-1',
    })
    
   Results in a 546 error code and:
   

>    shutdown (reason: Memory limit reached)
>    
>    "memory_used": [
>     {
>       "external": 359832,
>       "heap": 153877568,
>       "total": 154237400
>     }
>   ],

**Comment by zlotnika** [comment]
I'm running out of memory just doing a download of a storage object (under the 50mb limit). Moving around storage objects seems like an important standard use-case for edge functions to me.
